include/oneapi/dpl/internal/inclusive_scan_by_segment_impl.h

```diff
@@ -83,20 +83,16 @@ inclusive_scan_by_segment_impl(Policy&& policy, InputIterator1 first1, InputIter
     typedef typename ::std::iterator_traits<InputIterator2>::value_type ValueType;
     typedef typename ::std::decay<Policy>::type policy_type;
 
-    ValueType initial_value;
-    {
-        auto first2_acc = internal::get_access<sycl::access::mode::read>(policy, first2);
-        initial_value = first2_acc[0];
-    }
-
     const auto n = ::std::distance(first1, last1);
     // Check for empty and single element ranges
     if (n <= 0)
         return result;
-    else if (n == 1)
+
+    const ValueType initial_value = internal::get_data_at<ValueType>(policy, first2, 0);
+
+    if (n == 1)
     {
-        auto result_acc = internal::get_access<sycl::access::mode::write>(policy, result);
-        result_acc[0] = initial_value;
+        internal::set_data_at(policy, result, 0, initial_value);
```

Comment:
Considering this line 95 and line 91 we have following semantic: `*result = *first2`.
In case of the device allocations we have some copying overhead: "device->host->device". I believe we can avoid that, f.e., by using sycl::queue::copy method which is optimized for direct "device-to-device"  copying.

Reply:
Fixed

Reply:
Fixed